### PR TITLE
Tests for ipa-ipa migration tool

### PR DESCRIPTION
**Pull request by menonsudhir**
This patch includes tests for ipa-ipa migration tool.

### CODE REVIEW

File: ipatests/test_integration/test_ipa_ipa_migration.py [hunk 1]
```diff
@@ -0,0 +1,762 @@
...
+            bind_pwd=None,
+            extra_args=param,
+        )
+        install_msg = self.replicas[0].get_file_contents(
+            paths.IPA_MIGRATE_LOG, encoding="utf-8"
+        )
+        assert DEBUG_MSG in install_msg
+        assert result.returncode == 0
+
+   def test_ipa_migrate_using_db_ldif(self):
```
Comment (flo-renaud) [inline ipatests/test_integration/test_ipa_ipa_migration.py:694]: ipatests/test_integration/test_ipa_ipa_migration.py:694: [E0001(syntax-error), ] Parsing failed: 'unindent does not match any outer indentation level (<unknown>, line 694)')

**Comment by mreynolds389** [comment]
This looks really good!  I think eventually we need more tests to verify if each type of "entry" is migrated:  DNA ranges, HBAC, Sudo Rules, etc.  But that can be done later (or new PR).

**Review by mreynolds389 — APPROVED** [review]

**Comment by menonsudhir** [comment]
> This looks really good! I think eventually we need more tests to verify if each type of "entry" is migrated: DNA ranges, HBAC, Sudo Rules, etc. But that can be done later (or new PR).

@mreynolds389 yes thats the plan i am going to add those tests as well, which captures HBAC, SUDO, SID etc in a seperate PR

**Comment by flo-renaud** [comment]
@menonsudhir Can you mark the tests test_ipa_migrate_stage_mode and test_ipa_migrate_version_option as xfail?
You need to add
```
    @pytest.mark.xfail(reason="https://issues.redhat.com/browse/RHEL-45463", strict=True)
    def test_ipa_migrate_stage_mode(self, empty_log_file):
...
    @pytest.mark.xfail(reason="https://issues.redhat.com/browse/RHEL-46003", strict=True)
    def test_ipa_migrate_version_option(self):
```

You also need to add the new test in the prci definitions in nightly_latest_selinux.yaml, nightly_latest_testing_selinux.yaml, nightly_latest_testing.yaml, nightly_latest.yaml, nightly_previous.yaml, nightly_rawhide.yaml.

**Comment by flo-renaud** [comment]
@menonsudhir thanks for the update. You can remove the temp commit.

For future ref, link to green run: `temp_commit` [Details](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/d759de48-3a0e-11ef-8be1-fa163e929191)

**Comment by flo-renaud** [comment]
master:

* 5fe7cf50a98b7d37f24a900d506249611b9fd241 ipatests: Tests for ipa-ipa migration tool

**Pull request closed without merging**